doc/sai-ptf/pcbb_test_plan.md

```diff
@@ -0,0 +1,346 @@
+  | ipv6's falls in fc02::60:0  |   ipv6's falls in fc00:1:: |
+
+### Testing Data Packet <!-- omit in toc --> 
+
+This test should cover the data in the table below
+
+DSCP_MAP_TABLE
+|DSCP|TC to verify| Expected DSCP after encap(outer)|Outgoing Queue(Tunnel)|Priority(Port)|
+| ---- | ---- | --- |-|-|
+|8|0|8|0|0|
+|0-1|1|0|1|0|
+|3|3|2|2|3|
+|4|4|6|6|4|
+|5-32|1|0|1|0|
+|33|8|33|1|0|
+|34-45|1|0|1|0|
+|47|1|0|1|0|
+|46|5|46|5|0|
+|48|7|48|7|7|
+|49-63|1|0|1|0|
+
+
+For ECN testing
+
+ECN_NON_CGN_TABLE:
+|DSCP| ECN Outer DSCP |Outer ECN |Inner DSCP |Inner ECN|
+|-|-|-|-|-|
+|33|3|33|3|33|3|
+|33|2|33|2|33|2|
+|33|1|33|1|33|1|
+|3|3|2|3|3|3|
+|3|2|2|2|3|2|
+|3|1|2|1|3|1|
+|4|3|6|3|4|3|
+|4|2|6|2|4|2|
+|4|1|6|1|4|1|
+
+ECN_CGN_TABLE:
+|DSCP|ECN|Outer DSCP|Outer ECN|Inner DSCP|Inner ECN|
+|-|-|-|-|-|-|
+|3|3|2|3|3|3|
+|3|2|2|3|3|3|
+|3|1|2|3|3|3|
+|4|3|6|3|4|3|
+|4|2|6|3|4|3|
+|4|1|6|3|4|3|
+
+
+#### IPV4 IN IPV4 Packet <!-- omit in toc --> 
+- expected egress encap packet=Ether(dst=ROUTER_MAC)/IP(dst=``ACTIVE_TOR_IP``,src=`STAND_BY_TOR_IP``,ip_dscp=``EXP_DSCP COL``, ip_ecn=``OUTER_ECN``)/IP(dst=192.168.60.1,src=192.168.1.1, ip_dscp=``DSCP COL``, ip_ecn=``INNER_ECN``)/TCP()
+- ingress packet = Ether(dst=00:01:01:01:02:a0,src=ROUTER_MAC)/IP(dst=192.168.60.1,src=192.168.1.1, ip_dscp=``DSCP COL``, ip_ecn=``INNER_ECN``)/TCP()
+
+#### IPV6 IN IPV4 Packet <!-- omit in toc --> 
+- expected egress encap packet=Ether(dst=ROUTER_MAC)/IP(dst=``ACTIVE_TOR_IP``,src=`STAND_BY_TOR_IP``,ip_dscp=``EXP_DSCP COL``, ip_ecn=``OUTER_ECN``)/IP(fc02::60:1,src=fc02::1:1,ip_dscp=``EXP_DSCP COL``, ip_ecn=``INNER_ECN``)/TCP()
+- ingress packet = Ether(dst=00:01:01:01:02:a0,src=ROUTER_MAC)/IP(dst=``REMOTE_SERVER_IPv6``,src=fc02::1:1,ip_dscp=``EXP_DSCP COL``, ip_ecn=``INNER_ECN``)/TCP()
+
+
+### Test steps: <!-- omit in toc --> 
+- encap_dscp_remap:
+
+1. Make sure LAGs and NextHop groups set as basic [config](./config_data/config_t0.md) 
+2. Make sure tunnel DSCP in PIPE mode, port and tunnel binding to the DSCP map, queue map, and priority map as basic [config](./config_data/config_t0.md)  
+3. Generate packet, take one row from the ``DSCP_MAP_TABLE``, set the ``ip_dscp`` accordingly
+4. Send input packet from port1.
+5. Create the expected ipinip packet with ``Inner DSCP`` and ``Outer dscp`` according to ``DSCP_MAP_TABLE``. 
+6. Recieve ipinip packet from any lag1-4 member port. Compare it with the expected ipinip packet.
+
+- encap_dscp_queue:
+
+1. Make sure LAGs and NextHop groups set as basic [config](./config_data/config_t0.md) 
+2. Make sure tunnel DSCP in PIPE mode, port and tunnel binding to the DSCP map, queue map, and priority map as basic [config](./config_data/config_t0.md)  
+3. According to DSCP_MAP_TABLE check the corrosponding queue's packets stats on the possible ports(use sai_thrift_get_queue_stats with "SAI_QUEUE_STAT_PACKETS")
+4. Generate 1000 packets, take one row from the ``DSCP_MAP_TABLE``, set the ``ip_dscp`` accordingly
```

Comment:
For SAI test, it's not necessary to generate 1000 packets, because there is noise from background packets.